### PR TITLE
Object detection tutorials

### PR DESCRIPTION
**Pull request by pkranen**
Added a tutorial for Faster R-CNN and a separate tutorial for Fast R-CNN using the new code. Restricted the old Fast R-CNN tutorial to BrainScript.

### CODE REVIEW

**Comment by sankethka** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/Object-Detection-using-Fast-R-CNN-BrainScript.md](https://github.com/MicrosoftDocs/cognitive-toolkit-docs/blob/pkranen/odBlogs/articles/Object-Detection-using-Fast-R-CNN-BrainScript.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cognitive-toolkit/object-detection-using-fast-r-cnn-brainscript?branch=pr-en-us-109) |
[articles/Object-Detection-using-Fast-R-CNN.md](https://github.com/MicrosoftDocs/cognitive-toolkit-docs/blob/pkranen/odBlogs/articles/Object-Detection-using-Fast-R-CNN.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cognitive-toolkit/object-detection-using-fast-r-cnn?branch=pr-en-us-109) |
[articles/Object-Detection-using-Faster-R-CNN.md](https://github.com/MicrosoftDocs/cognitive-toolkit-docs/blob/pkranen/odBlogs/articles/Object-Detection-using-Faster-R-CNN.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cognitive-toolkit/object-detection-using-faster-r-cnn?branch=pr-en-us-109) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C8%5C30%5C104c4e94-72b4-a5b2-e54a-b355635eb6e8%5CPullRequest%5C201708301503191003-109%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 1]
```diff
@@ -0,0 +1,537 @@
+---
+title:   Object  detection using Fast R-CNN
+author:    pkranen
+ms.author:   pkranen
+ms.date:   06/27/2017
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >06/27/2017 [](start = 11, length = 10)

update

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+ms.custom:   cognitive-toolkit
+ms.topic:   conceptual
+ms.service:  Cognitive-services
+ms.devlang:   brainscript
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >brainscript [](start = 14, length = 11)

?

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+---
+
+# Object  detection using Fast R-CNN
+
+**Table of Contents**
+
+- [Summary](#summary)
+- [Setup](#setup)
+- [Run the toy example](#run-the-toy-example)
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >toy [](start = 11, length = 3)

toy or test? I don't see any toys

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+## Summary
+
+<p align="center">
+<a target="_blank" href="./Tutorial_FastRCNN/svm_4WIN_20160803_12_37_07_Pro.jpg"><img src="./Tutorial_FastRCNN/svm_4WIN_20160803_12_37_07_Pro.jpg" alt="image" height="250"/></a>
+<a target="_blank" href="./Tutorial_FastRCNN/bus_01.jpg"><img src="./Tutorial_FastRCNN/bus_01.jpg" alt="image" height="250"/></a>
+</p>
+
+**This tutorial describes how to use CNTK Fast R-CNN with BrainScript and cntk.exe. Fast R-CNN using the CNTK Python API is described [here](./Object-Detection-using-Fast-R-CNN.md).**
+
+The above are examples images and object annotations for the grocery data set (left) and the Pascal VOC data set (right) used in this tutorial.
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >(left) and the Pascal VOC data set (right) [](start = 78, length = 42)

left and right are incorrect on smaller screens, then it is top or bottom.... 

perhaps (refrigerator image) and (street view) ???

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 5]
```diff
@@ -0,0 +1,537 @@
...
+The paper is accepted to ICCV 2015, and archived at https://arxiv.org/abs/1504.08083.
+Fast R-CNN builds on previous work to efficiently classify object proposals using deep convolutional networks.
+Compared to previous work, Fast R-CNN employs a _region of interest pooling_ scheme that allows to reuse the computations from the convolutional layers.
+
+Additional material: a detailed tutorial for object detection using CNTK Fast R-CNN with BrainScript 
+(including optional SVM training and publishing the trained model as a Rest API) can be found [here](https://github.com/Azure/ObjectDetectionUsingCntk).
+
+## Setup
+
+To run the code in this example, you need a CNTK Python environment (see [here](./Setup-CNTK-on-your-machine.md) for setup help). You need to work from your Python 3.4 environment (if you are using Anaconda Python type `activate cntk-py34` (from a standard command line, not PowerShell), assuming `cntk-py34` is your environment name). Further you need to install a few additional packages. From your Python 3.4 environment (64bit version assumed, Python 3.5 analogously), go to the FastRCNN folder and run:
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >3.5 [](start = 455, length = 3)

above you say you need 3.4?

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 6]
```diff
@@ -0,0 +1,537 @@
...
+
+```
+pip install -r requirements.txt
+```
+
+Known issue: to install scikit-learn you might have to run `conda install scikit-learn` if you use Anaconda Python.
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >Known issue: [](start = 0, length = 12)

i'm confused how to set this up... 

is 'requirement.txt now sufficient or do I need all this?
numpy and scipy are cntk pre-req ... why would I need them. these are in my environment already ...

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 7]
```diff
@@ -0,0 +1,537 @@
...
+  * scikit_image-0.12.3-cp35-cp35m-win_amd64.whl
+  * opencv_python-3.2.0-cp35-cp35m-win_amd64.whl
+
+Once you download the respective wheel binaries, install them with (Python 3.5 analogously):
+
+```
+pip install your_download_folder/scikit_image-0.12.3-cp34-cp34m-win_amd64.whl
+```
+
+Remark 1: if you see the message _No module named past_ when running the scripts please execute `pip install future`.
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >Remark 1:  [](start = 0, length = 10)

[!NOTE]

see for example: [https://docs.microsoft.com/en-us/cognitive-toolkit/Setup-Windows-Binary-Script](https://docs.microsoft.com/en-us/cognitive-toolkit/Setup-Windows-Binary-Script)

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 8]
```diff
@@ -0,0 +1,537 @@
...
+<br>Remark 2: If you have a Python 3.5 environment you need the corresponding _cp35_ wheels.
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: ><br> [](start = 0, length = 4)

we should avoid adding html elements

Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >If you have a Python 3.5 environment you need the corresponding _cp35_ wheels. [](start = 14, length = 78)

I don't understand this - well, I do - but this is awkward ...this whole handling of Python 3.5 is very inconcistent in the paragraphs above (sometimes it is explicitly done, sometimes just mentioned ....)

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 9]
```diff
@@ -0,0 +1,537 @@
...
+
+This tutorial code assumes you are using 64bit version of Python, either 3.4 or 3.5 on Windows or 3.4 on Linux, since the required Fast R-CNN DLL files under [utils](https://github.com/Microsoft/CNTK/tree/master/Examples/Image/Detection/FastRCNN/BrainScript/fastRCNN/utils)
+are prebuilt for those versions. If your task requires the use of a different Python version, please recompile these DLL files yourself in the correct environment (see [below](#pre-compiled-binaries-for-bounding-box-regression-and-non-maximum-suppression)).
+
+The tutorial further assumes that **the folder where cntk.exe resides is in your PATH environment variable**. (To add the folder to your PATH you can run the following command from a command line (assuming the folder where cntk.exe is on your machine is C:\src\CNTK\x64\Release): `set PATH=C:\src\CNTK\x64\Release;%PATH%`.)
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >The tutorial further assumes that **the folder where cn [](start = 0, length = 55)

isn't that whole thing part of the python setup as well as the activation?

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 10]
```diff
@@ -0,0 +1,537 @@
...
+
+### Pre-compiled binaries for bounding box regression and non maximum suppression
+
+The folder `Examples\Image\Detection\FastRCNN\BrainScript\fastRCNN\utils` contains pre-compiled binaries that are required for running Fast R-CNN. They versions that are currently contained in the repository are Python 3.4 and 3.5 for Windows and Python 3.4 for Linux, all 64 bit. If you need a different version you can compile it following these steps:
+
+* `git clone --recursive https://github.com/rbgirshick/fast-rcnn.git`
+* `cd $FRCN_ROOT/lib`
+* `make`
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: I believe this should be in one section guarded by 
```
```

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 11]
```diff
@@ -0,0 +1,537 @@
...
+are prebuilt for those versions. If your task requires the use of a different Python version, please recompile these DLL files yourself in the correct environment (see [below](#pre-compiled-binaries-for-bounding-box-regression-and-non-maximum-suppression)).
+
+The tutorial further assumes that **the folder where cntk.exe resides is in your PATH environment variable**. (To add the folder to your PATH you can run the following command from a command line (assuming the folder where cntk.exe is on your machine is C:\src\CNTK\x64\Release): `set PATH=C:\src\CNTK\x64\Release;%PATH%`.)
+
+### Pre-compiled binaries for bounding box regression and non maximum suppression
+
+The folder `Examples\Image\Detection\FastRCNN\BrainScript\fastRCNN\utils` contains pre-compiled binaries that are required for running Fast R-CNN. They versions that are currently contained in the repository are Python 3.4 and 3.5 for Windows and Python 3.4 for Linux, all 64 bit. If you need a different version you can compile it following these steps:
+
+* `git clone --recursive https://github.com/rbgirshick/fast-rcnn.git`
+* `cd $FRCN_ROOT/lib`
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >$FRCN_ROOT [](start = 6, length = 10)

where does this come from?

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 12]
```diff
@@ -0,0 +1,537 @@
...
+* `make`
+  * Alternatively you can run `python setup.py build_ext --inplace`. On Windows you might have to comment out the extra compile args in lib/setup.py:
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >`python setup.py build_ext --inplace` [](start = 30, length = 37)

what? where? alternativ to make or what?

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 13]
```diff
@@ -0,0 +1,537 @@
...
+        #extra_compile_args=["-Wno-cpp", "-Wno-unused-function"],
+    )
+  ]
+  ```
+
+* copy the generated `cython_bbox` and `cython_nms` binaries from `$FRCN_ROOT/lib/utils` to `$CNTK_ROOT/Examples/Image/Detection/fastRCNN/utils`.
+
+### Example data and baseline model
+
+We use a pre-trained AlexNet model as the basis for Fast-R-CNN training. The pre-trained AlexNet is available at [https://www.cntk.ai/Models/AlexNet/AlexNet.model](https://www.cntk.ai/Models/AlexNet/AlexNet.model). Please store the model at `<cntkroot>/PretrainedModels`. To download the data please run
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: ><cntkroot> [](start = 242, length = 10)

is this the same as $CNTK_ROOT? 
let's be consistent ...

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 14]
```diff
@@ -0,0 +1,537 @@
...
+
+```
+python install_grocery.py
+```
+
+from the `Examples/Image/DataSets/Grocery` folder. 
+
+## Run the toy example
+
+In the toy example we train a CNTK Fast  R-CNN model to detect grocery items in a refrigerator.
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >  [](start = 39, length = 1)

one blank?

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 15]
```diff
@@ -0,0 +1,537 @@
...
+<a target="_blank" href="./Tutorial_FastRCNN/nn_0WIN_20160803_11_28_42_Pro.jpg"><img src="./Tutorial_FastRCNN/nn_0WIN_20160803_11_28_42_Pro.jpg" alt="image" height="200"/></a>
+<a target="_blank" href="./Tutorial_FastRCNN/nn_1WIN_20160803_11_42_36_Pro.jpg"><img src="./Tutorial_FastRCNN/nn_1WIN_20160803_11_42_36_Pro.jpg" alt="image" height="200"/></a>
+<a target="_blank" href="./Tutorial_FastRCNN/nn_2WIN_20160803_11_46_03_Pro.jpg"><img src="./Tutorial_FastRCNN/nn_2WIN_20160803_11_46_03_Pro.jpg" alt="image" height="200"/></a>
+<a target="_blank" href="./Tutorial_FastRCNN/nn_3WIN_20160803_11_48_26_Pro.jpg"><img src="./Tutorial_FastRCNN/nn_3WIN_20160803_11_48_26_Pro.jpg" alt="image" height="200"/></a>
+<a target="_blank" href="./Tutorial_FastRCNN/nn_4WIN_20160803_12_37_07_Pro.jpg"><img src="./Tutorial_FastRCNN/nn_4WIN_20160803_12_37_07_Pro.jpg" alt="image" height="200"/></a>
+</p>
+
+### Step details
+
+__A1:__ The script first `A1_GenerateInputROIs.py` generates ROI candidates for each image using [selective search](#selective-search).
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >script first [](start = 12, length = 12)

first script

?

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 16]
```diff
@@ -0,0 +1,537 @@
...
+It then stores them in a [CNTK Text Format](./BrainScript-CNTKTextFormat-Reader.md) as input for `cntk.exe`.
+Additionally the required CNTK input files for the images and the ground truth labels are generated.
+The script generates the following folders and files under the `FastRCNN` folder:
+
+* `proc` - root folder for generated content.
+  * `grocery_2000` - contains all generated folders and files for the `grocery` example using `2000` ROIs. If you run again with a different number of ROIs the folder name will change correspondingly.
+    * `rois` - contains the raw ROI coordinates for each image stored in text files.
+    * `cntkFiles` - contains the formatted CNTK input files for images (`train.txt` and `test.txt`), ROI coordinates (`xx.rois.txt`) and ROI labels (`xx.roilabels.txt`) for `train` and `test`. (Format details are provided [below](#cntk-input-file-format).)
+
+All parameters are contained in `PARAMETERS.py`, for example `cntk_nrRois = 2000` to set the number of ROIs used for training and testing. We describe parameters in the section [Parameters](#parameters) below.
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >  [](start = 60, length = 1)

isn;t here a verb missing? "change"?

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 17]
```diff
@@ -0,0 +1,537 @@
...
+
+__A2:__ The script `A2_RunWithBSModel.py` runs cntk using cntk.exe and a BrainScript config file ([configuration details](#cntk-configuration)).
+A script that uses the new CNTK Python API for Fast R-CNN training will be added soon.
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: not sure how this sentence is helping anyone?

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 18]
```diff
@@ -0,0 +1,537 @@
...
+
+* `B1_VisualizeInputROIs.py` visualizes the candidate input ROIs.
+* `B2_EvaluateInputROIs.py` computes the recall of the ground truth ROIs with respect to the candidate ROIs.
+* `B3_VisualizeOutputROIs.py` visualize the bounding boxes and predicted labels.
+
+## Run Pascal VOC
+
+The [Pascal VOC](http://host.robots.ox.ac.uk/pascal/VOC/) (PASCAL Visual Object Classes) data is a well known set of standardised images for object class recognition.
+Training or testing CNTK Fast R-CNN on the Pascal VOC data requires a GPU with at least 4GB of RAM.
+Alternatively you can run using the CPU, which will however take _some_ time.
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: > take _some_ time. [](start = 59, length = 18)

even GPU will take _some_ time ....

Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: > take _some_ time. [](start = 59, length = 18)

is this a word? <drop it>

File: articles/Object-Detection-using-Fast-R-CNN.md [hunk 19]
```diff
@@ -1,12 +1,12 @@
 ---
 title:   Object  detection using Fast R-CNN
-author:    chrisbasoglu
-ms.author:   cbasoglu
+author:    pkranen
+ms.author:   pkranen
 ms.date:   06/27/2017
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN.md]: >06/27/2017 [](start = 11, length = 10)

date

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md (hunk 9 above)
Comment (pkranen) [reply articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: We had numerous people asking about issues related to 'cannot find cntk.exe' or the like. That's why we added this sentence in bold to the setup section (months ago btw).

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md (hunk 11 above)
Comment (pkranen) [reply articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: This is the notation used here [https://github.com/rbgirshick/fast-rcnn#installation-sufficient-for-the-demo](https://github.com/rbgirshick/fast-rcnn#installation-sufficient-for-the-demo). The steps are just copy-pasted to our setup section for convenience.

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 20]
```diff
@@ -0,0 +1,537 @@
...
+The node names `network.features`, `network.conv5_y` etc. can be derived from looking at the log output of the cntk.exe call
+(contained in the log output of the `A2_RunWithBSModel.py` script).
+
+The model definition(`model (features, rois) = ...`) first normalizes the features by subtracting 114 for each channel and pixel.
+Then the normalized features are pushed through the `convLayers` followed by the `ROIPooling` and finally the `fcLayers`.
+The output shape (width:height) of the ROI pooling layer is set to `(6:6)` since this is the shape nd size that the
+pre-trained `fcLayers` from the AlexNet model expect. The output of the `fcLayers` is fed into a dense layer that
+predicts one value per label (`NumLabels`) for each ROI.
+
+The following six lines define the input: an image of size 1000 x 1000 x 3 (`$ImageH$:$ImageW$:$ImageC$`),
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >an i [](start = 42, length = 4)

bullet point list ?

Perhaps?

File: articles/Object-Detection-using-Fast-R-CNN-BrainScript.md [hunk 21]
```diff
@@ -0,0 +1,537 @@
...
+
+<p align="center">
+<a target="_blank" href="./Tutorial_FastRCNN/WIN_20160803_11_29_07_Pro.noGridNoFiltering.roi.jpg"><img src="./Tutorial_FastRCNN/WIN_20160803_11_29_07_Pro.noGridNoFiltering.roi.jpg" alt="image" height="300"/></a>
+<a target="_blank" href="./Tutorial_FastRCNN/WIN_20160803_11_29_07_Pro.noGrid.roi.jpg"><img src="./Tutorial_FastRCNN/WIN_20160803_11_29_07_Pro.noGrid.roi.jpg" alt="image" height="300"/></a>
+<a target="_blank" href="./Tutorial_FastRCNN/WIN_20160803_11_29_07_Pro.roi.jpg"><img src="./Tutorial_FastRCNN/WIN_20160803_11_29_07_Pro.roi.jpg" alt="image" height="300"/></a>
+</p>
+
+To complement the detected ROIs from Selective Search, we add ROIs that uniform cover the image at different scales and aspect ratios.
+The image on the left shows an example output of Selective Search, where each possible object location is visualized by a green
+rectangle. ROIs that are too small, too big, etc. are discarded (middle) and finally ROIs that uniformly cover the image are added (right).
```
Comment (wolfma61) [inline articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: >right [](start = 132, length = 5)

it seems like middle left right etc are depending on your screen resolution?

Comment (pkranen) [reply articles/Object-Detection-using-Fast-R-CNN-BrainScript.md]: replaced by first, second and third to account for resolution

**Comment by sankethka** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/Object-Detection-using-Fast-R-CNN-BrainScript.md](https://github.com/MicrosoftDocs/cognitive-toolkit-docs/blob/pkranen/odBlogs/articles/Object-Detection-using-Fast-R-CNN-BrainScript.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cognitive-toolkit/object-detection-using-fast-r-cnn-brainscript?branch=pr-en-us-109) |
[articles/Object-Detection-using-Fast-R-CNN.md](https://github.com/MicrosoftDocs/cognitive-toolkit-docs/blob/pkranen/odBlogs/articles/Object-Detection-using-Fast-R-CNN.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cognitive-toolkit/object-detection-using-fast-r-cnn?branch=pr-en-us-109) |
[articles/Object-Detection-using-Faster-R-CNN.md](https://github.com/MicrosoftDocs/cognitive-toolkit-docs/blob/pkranen/odBlogs/articles/Object-Detection-using-Faster-R-CNN.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cognitive-toolkit/object-detection-using-faster-r-cnn?branch=pr-en-us-109) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C9%5C5%5C104c4e94-72b4-a5b2-e54a-b355635eb6e8%5CPullRequest%5C201709052256108505-109%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**